File: SocratesFr/CandidateManagement/Room.cs
﻿using System;
using System.ComponentModel;

namespace SocratesFr.CandidateManagement
{
    public class Room
    {
        private RoomType roomType;

        public enum RoomType
        {
            SINGLE = 0,
            DOUBLE,
            TRIPLE,
            NO_ACCOMODATION
        }

        public Room(RoomType room)
        {
            this.roomType = room;

            switch (roomType)
            {
                case RoomType.SINGLE:
                    Price = 610;
                    break;
                case RoomType.DOUBLE:
                    Price = 510;
                    break;
                case RoomType.TRIPLE:
                    Price = 410;
                    break;
                case RoomType.NO_ACCOMODATION:
                    Price = 240;
                    break;
                default:
                    throw new InvalidEnumArgumentException("Please select one of the four room price.");
            }
        }

        public object Price { get; private set; }
    }
}

File: src/main/java/Room.java
/**
 * Created by lenovo_3 on 22/01/2018.
 */
public enum Room {
    SINGLE(610),
    DOUBLE(510),
    TRIPLE(410);

    private final int price;

    Room(int price) {
        this.price = price;
    }

    public int price() {
        return price;
    }
}
